kbl2vec/src/main/java/com/foursoft/harness/kbl2vec/transform/geometry/geo_3d/BuildingBlockSpecification3DTransformer.java

```diff
@@ -63,6 +60,15 @@ public TransformationResult<VecBuildingBlockSpecification3D> transform(final Tra
                 .withDownstream(KblSegment.class, VecGeometrySegment3D.class,
                                 Query.fromLists(source.getParentKBLContainer().getSegments()),
                                 VecBuildingBlockSpecification3D::getGeometrySegments)
+                .withDownstream(ConnectionOrOccurrence.class, VecOccurrenceOrUsageViewItem3D.class,
+                                () -> getPlaceableOccurrences(source),
+                                VecBuildingBlockSpecification3D::getPlacedElementViewItem3Ds)
                 .build();
     }
+
+    private List<ConnectionOrOccurrence> getPlaceableOccurrences(final KblHarness kblHarness) {
+        return kblHarness.getConnectionOrOccurrences().stream()
+                .filter(HasPlacement.class::isInstance)
+                .toList();
+    }
```

Comment:
This is repeated code and could. For this we have a `Queries` class. In addition, if write the method like this, you get rid of the ugly lambda syntax in linn 64. Same applies to BBS2DTransformer

```java
    public static Query<ConnectionOrOccurrence> partOccurrences(final List<ConnectionOrOccurrence> components) {
        return () -> components
                .stream()
                .filter(c -> !(c instanceof KblConnection))
                .toList();
    }
```